pkg/installation/util.go

```diff
@@ -25,23 +25,29 @@ import (
 	"sigs.k8s.io/krew/pkg/installation/receipt"
 )
 
-func getDownloadTarget(index index.Plugin) (version, sha256sum, uri string, fos []index.FileOperation, bin string, err error) {
+func getDownloadTarget(plugin index.Plugin) (version, sha256sum, uri string, fos []index.FileOperation, bin string, err error) {
 	// TODO(ahmetb): We have many return values from this method, indicating
 	// code smell. More specifically we return all-or-nothing, so ideally this
 	// should be converted into a struct, like InstallOperation{} contains all
 	// the data needed to install a plugin.
-	p, ok, err := GetMatchingPlatform(index.Spec.Platforms)
+	p, ok, err := GetMatchingPlatform(plugin.Spec.Platforms)
 	if err != nil {
 		return "", "", "", nil, p.Bin, errors.Wrap(err, "failed to get matching platforms")
 	}
 	if !ok {
 		return "", "", "", nil, p.Bin, errors.New("no matching platform found")
 	}
-	version = index.Spec.Version
+	version = plugin.Spec.Version
 	uri = p.URI
 	sha256sum = p.Sha256
+
+	files := p.Files
+	if len(files) == 0 {
```

Comment:
I'm removing this method (`getDownloadTarget`) soon as part of #293.

I think we should consider a method like `func applyDefaults(*index.Platform)` to achieve defaulting on the FileOperations. Are you open to such a method, that's easily testable etc?

Reply:
In principle, I like the idea. However, I didn't find a good spot for this. The best place is close to validation, but there we pass in a copy of the plugin object which loses any mutations. And I didn't want to put this into `ReadPluginFile`.

So I ended up, putting it in `moveToInstallDir`. Not ideal, though..

Reply:
I think that's why we should call `applyDefaults(&plugin)` right before we do moveToInstallDir. Then we can do  `moveToInstallDir(..., plugin.Files)`.

Don't move inside `moveToInstallDir` as the name indicates, it's not expected to do defaulting.

Reply:
Alright, I moved the logic one level up into `install`. Also, I added a unit test for `applyDefaults`.